include/solvers/post_processors_smoothing.h

```diff
@@ -0,0 +1,181 @@
+
+#ifndef lethe_post_processors_smoothing_h
+#define lethe_post_processors_smoothing_h
+
+// DEALII INCLUDES
+// Base
+#include <deal.II/base/tensor.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/fe_values_extractors.h>
+#include <deal.II/fe/mapping_fe.h>
+#include <deal.II/fe/mapping_q.h>
+
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/la_parallel_vector.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/solver_control.h>
+#include <deal.II/lac/sparsity_tools.h>
+#include <deal.II/lac/trilinos_parallel_block_vector.h>
+#include <deal.II/lac/trilinos_precondition.h>
+#include <deal.II/lac/trilinos_solver.h>
+#include <deal.II/lac/trilinos_sparse_matrix.h>
+#include <deal.II/lac/trilinos_vector.h>
+
+// Numerics
+#include <deal.II/distributed/solution_transfer.h>
+
+#include <deal.II/numerics/data_postprocessor.h>
+
+// Rheological models
+#include <core/rheological_model.h>
+
+#include <solvers/simulation_parameters.h>
+
+
+// standard library includes includes
+#include <vector>
+
+using namespace dealii;
+
+
+/**
+ * A base class that holds the dof_handler, fe_values and the simulation output
+ * to calculate post-processed parameters
+ *
+ * @tparam dim An integer that denotes the dimension of the space in which
+ * the flow is solved
+ *
+ * @tparam triangulation Flow triangulation
+ * @tparam simulation_parameters The simulation parameters
+ * @tparam number_quadrature_points The number of quadrature points
+ * @tparam mpi_communicator Allowing communication between cores
+ */
+
+template <int dim, typename VectorType>
+class PostProcessorSmoothing
+{
+public:
+  // Member functions
+  PostProcessorSmoothing(
+    std::shared_ptr<parallel::DistributedTriangulationBase<dim>> triangulation,
```

Comment:
If you don't need to store it locally, I would pass isntead just a reference to the DistributedTriangulationBase instead of passing a shared ptr. We are trying to have cleaner interfaces this way :).

Reply:
Good catch, it is changed.